docs/additional/internal_aspects_de.md

```diff
@@ -0,0 +1,313 @@
+Zu beachten ist, dass die `dogu-build-lib` von der Google-Cloud abhängig ist.
+Sie ist in der Lage, Instanzen des Cloudogu EcoSystems in der Cloud zu provisionieren, damit diese für
+Integrationstests genutzt werden können.
+
+Im Folgenden werden die verschiedenen Methoden, die in einer Pipeline zum Testen verwendet werden können, beschrieben.
+Dabei gehen wir speziell auf die Unterstützung unserer Bibliotheken ein.
+
+[ces-build-lib]: https://github.com/cloudogu/ces-build-lib
+[dogu-build-lib]: https://github.com/cloudogu/dogu-build-lib
+
+##### Shell-Tests & Linting
+
+Mit der Entwicklung von Dogus werden voraussichtlich mehrere Bash-Skripte erstellt.
+Diese sollten getestet und auf Smells überprüft werden.
+Für das Linting wird der Einsatz der `ShellCheck()` Methode aus der `dogu-build-lib` empfohlen.
+Ein beispielhafter Aufruf sieht in einem Jenkins-File wie folgt aus:
+
+```groovy
+stage('Shellcheck') {
+    shellCheck("resources/startup.sh")
+}
+```
+
+Für das Testing kann [Bats][bats] aus der `ces-build-lib` verwendet werden:
+
+```groovy
+Docker docker = new Docker(this)
+
+stage('Bats Tests') {
+    Bats bats = new Bats(this, docker)
+    bats.checkAndExecuteTests()
+}
+```
+
+[bats]: https://github.com/bats-core/bats-core
+
+##### Dockerfile-Linting
+
+Das Dockerfile sollte nach Best Practices aufgebaut und natürlich auch syntaktisch korrekt sein.
+Dafür wird der Einsatz der `lintDockerfile()` Methode aus der `ces-build-lib` empfohlen.
+Ein beispielhafter Aufruf sieht in einem Jenkins-File wie folgt aus:
+
+```groovy
+stage('Lint') {
+    lintDockerfile()
+}
+```
+
+##### End-to-End-Tests (UI, API, CAS-Plugin)
+
+Die Absicht der End-to-End Tests liegt darin, den Mehrwert des Dogus im Cloudogu EcoSystem zu testen.
+Dabei bezieht man sich zum Beispiel auf die Integration des CAS-Dogus:
+
+- Front-Channel-Login
+- Back-Channel-Login
+- Front-Channel-Logout
+- Back-Channel-Logout
+- Änderung der Admin-Gruppe
+
+Die `dogu-build-lib` bietet hierbei verschiedene Methoden zur Ausführung.
+Ein Aufruf im Jenkins-File sieht wie folgt aus:
+
+```groovy
+stage('Integration tests') {
+    ecoSystem.runCypressIntegrationTests()
+}
+```
+
+Die Tests selbst müssen sich unter dem Startverzeichnis des Dogus im Ordner `integrationTests` befinden.
+Es existiert außerdem eine [Bibliothek][dogu-integration-lib], die zur
+Unterstützung bei der Implementierung der Tests dient.
+
+[dogu-integration-lib]: https://github.com/cloudogu/dogu-integration-test-library
+
+##### Trivy
+
+Es ist wichtig die Sicherheitslücken in einem Docker-Image, bzw. dem Zielsystem zu identifizieren und zu beheben.
+Dafür bietet sich der Scanner Trivy an, welcher Sicherheitslücken im System identifiziert und priorisiert.
+Die `dogu-build-lib` bietet dafür eine [Komponente][trivy] an, die wie folgt verwendet werden kann, um das Image eines
+Dogus zu scannen.
+
+```groovy
+EcoSystem ecoSystem = new EcoSystem(this, "gcloud-credentials-id", "ssh-credentials-id")
```

Comment:
Are these fake credential IDs? (If not why should we expose them to the public?)
Should these be more "fake-y" so the internal developer can expect them to be replaced with production credential IDs?